src/components/HomePage/RepositoryList/SortDropdown/index.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useCallback, useState } from "react";
+import "./SortDropdown.css";
+import { useDispatch, useSelector } from "react-redux";
+import { getData } from "../../../../api/githubData";
+import { RootState } from "../../../../redux/rootReducer";
+import {
+  fetchRepositoryFailure,
+  fetchRepositoryRequest,
+  fetchRepositorySuccess,
+  setSortValue,
+} from "../../../../redux";
+import { stat } from "fs";
+
+const SortDropdown = () => {
+  const dispatch = useDispatch();
+  const searchData = useSelector((state: RootState) => {
+    return state.search;
+  });
+  const sortData = useSelector((state: RootState) => {
+    return state.sort;
+  });
+
+  const changeHandler = useCallback(
+    async (e: React.ChangeEvent<HTMLSelectElement>) => {
+      let data = [];
+      dispatch(setSortValue(e.target.value));
+      switch (e.target.value) {
+        case "most stars":
+          try {
+            dispatch(fetchRepositoryRequest());
+            data = await getData(searchData.searchValue, "stars&order=desc");
+            dispatch(fetchRepositorySuccess(data));
+          } catch (error) {
+            dispatch(fetchRepositoryFailure(error.message));
+          }
```

Comment:
Should only catch cases here and then call individual handlers accordingly, for eg:
```js
const changeHandler = useCallback(() => {
  switch(value) {
    case 1:
      handler1();
      break;
    case 2:
      handler2();
      break;
  }
}, [handler1, handler2]);
```